### PR TITLE
Add replica configuration into database.yml

### PR DESCRIPTION
**Pull request by robinhandoko**
Jira : https://dekeo.atlassian.net/browse/SNOW-304
Tested on Kamino

### database.yml
Screen Shot 2022-11-21 at 10 56 32

### execute query
Screen Shot 2022-11-18 at 15 46 31

### CODE REVIEW

File: libraries/drivers_framework_rails.rb [hunk 1]
```diff
@@ -38,13 +38,14 @@ def database_yml(db_driver)
 
         database = db_driver.out
         deploy_environment = deploy_env
+        replica = node['deploy'][app['shortname']]['sidekiq_on_replica']
```
Comment (shkrt) [inline libraries/drivers_framework_rails.rb]: this is not only for sidekiq, let's give it more generic name, i.e `replica_db_settings`

File: templates/default/database.yml.erb [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <%=
   config = {}
   (['development', 'production'] + Array.wrap(@environment)).select(&:present?).uniq.each do |env|
-    config[env] = @database
+    config[env] = @database.merge(replica: @replica)
```
Comment (shkrt) [inline templates/default/database.yml.erb]: we need to not change original hash if there is no `@replica` variable or if it's nil

**Review by shkrt — COMMENTED** [review]
looks like changes for `database.yml` are missing

File: libraries/drivers_framework_rails.rb (hunk 1 above)
Comment (shkrt) [reply libraries/drivers_framework_rails.rb]: this comment is not addressed

Comment (robinhandoko) [reply libraries/drivers_framework_rails.rb]: I've update the variable name and I think we don't need to update the database.yml template.

Screen Shot 2022-11-21 at 10 59 49

Comment (shkrt) [reply libraries/drivers_framework_rails.rb]: I mean variable that we pull from stack settings, is called `sidekiq_on_replica`.

Comment (shkrt) [reply libraries/drivers_framework_rails.rb]: image

Comment (robinhandoko) [reply libraries/drivers_framework_rails.rb]: ah ok got it.. Thank you..

**Comment by praveendhawan** [comment]
@shkrt the base branch for this PR should be `chore/chef-12-aws` or `master`?

**Comment by shkrt** [comment]
> @shkrt the base branch for this PR should be `chore/chef-12-aws` or `master`?

base branch is `chore/chef-12-aws`

**Comment by praveendhawan** [comment]
@robinhandoko Can you please change it to correct base branch?

**Review by shkrt — APPROVED** [review]

**Pull request merged**